pkg/detectors/easyinsight/easyinsight.go

```diff
@@ -38,20 +40,22 @@ func (s Scanner) Keywords() []string {
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
 	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	keyMatches := keyPat.FindAllStringSubmatch(dataStr, -1)
```

Comment:
I think it's worthwhile to filter unique values for both key and ID prior to looping through both.

@zricethezav do you think this would benefit from creating a cartesian product of possible values?

Reply:
> @rgmz Aren't we doing this already with the nested loops?

I suppose so, once the unique filtering is added. I was thinking the manual checks for `key == id` are a bit awkward.